### DISCUSSION TITLE
Symfony/4 - Detect if a migration is already executed

### DISCUSSION THREAD

**Discussion by kAlvaro** [symfony/symfony#44530] [Q&A]
We have a custom version of the `migrations:execute` command to cope with our app having several PostgreSQL schemas.

In Symfony/3 I used `Doctrine\DBAL\Migrations\Version::isMigrated()` to determine if a given version had already been executed, in order to avoid double executions, but that method is gone in Symfony/4. Has the feature been moved elsewhere or it's gone for good?

**Comment by stof** [comment]
This is discussed in the wrong repo, as the code you talk about is not part of Symfony at all.

**Reply by kAlvaro** [reply]
My excuses.
I can't seem to be able to close or remove the question. If moderators are able, please do it for me.

**Reply by kAlvaro** [reply]
For what it's worth, it seems that you can use `\Doctrine\Migrations\Version\MigrationStatusCalculator::getNewMigrations()`. If your version belongs there (`hasMigration($version)`), that means it has not been executed yet.
